### PR TITLE
Add /identity

### PR DESCRIPTION
**Pull request by drasko**
Signed-off-by: Drasko DRASKOVIC <drasko.draskovic@gmail.com>

### CODE REVIEW

File: manager/api/resources.go [hunk 1]
```diff
@@ -15,6 +15,27 @@ type apiResponse interface {
 	empty() bool
 }
 
+// Identity
+type identityReq struct {
+	Key string `json:"key"`
+}
+
+type identityRes struct {
+	ClientID string `json:"id"`
```
Comment (mijicd) [inline manager/api/resources.go]: Rename to `ID`.

File: manager/api/resources.go [hunk 2]
```diff
@@ -15,6 +15,27 @@ type apiResponse interface {
...
+}
+
+func (res identityRes) headers() map[string]string {
+	return map[string]string{}
```
Comment (mijicd) [inline manager/api/resources.go]: We might consider returning it as a header to optimize the flow.

File: manager/api/transport.go [hunk 3]
```diff
@@ -124,6 +131,15 @@ func MakeHandler(svc manager.Service) http.Handler {
 	return r
 }
 
+func decodeIdentity(_ context.Context, r *http.Request) (interface{}, error) {
```
Comment (mijicd) [inline manager/api/transport.go:134]: The same remark applies to requests: utilise headers to pass the token and identity.

**Review by mijicd — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
Re-pushed. Please review.

File: manager/api/endpoint.go [hunk 4]
```diff
@@ -167,3 +167,18 @@ func canAccessEndpoint(svc manager.Service) endpoint.Endpoint {
 		return accessRes{}, nil
 	}
 }
+
+func identityEndpoint(svc manager.Service) endpoint.Endpoint {
+	return func(_ context.Context, request interface{}) (interface{}, error) {
+		req := request.(identityReq)
+
+		res := identityRes{}
```
Comment (mijicd) [inline manager/api/endpoint.go]: Move after if, and "inline":

```
res := identityRes{
  id: id
}
```

File: manager/api/resources.go [hunk 5]
```diff
@@ -15,6 +15,31 @@ type apiResponse interface {
...
+	key string
+}
+
+type identityRes struct {
+	id string
+}
+
+func (res identityRes) headers() map[string]string {
+	return map[string]string{
+		"Location": res.id,
```
Comment (mijicd) [inline manager/api/resources.go]: `Location` should be used to provide resource URL, absolute or relative (in our case only the relative ones were used). I would recommend using a custom header for this, for example `X-client-id`.

File: manager/swagger.yaml [hunk 6]
```diff
@@ -56,6 +56,24 @@ paths:
...
+        Retrieves the entity identifier given its secret key.
+      tags:
+        - identity
+      parameters:
+        - $ref: "#/parameters/Authorization"
+      responses:
+        200:
+          description: ID retrieved
+          headers:
+            Location:
```
Comment (mijicd) [inline manager/swagger.yaml]: Update according to the change recommended above.

**Review by mijicd — CHANGES_REQUESTED** [review]

File: manager/mocks/security.go [hunk 7]
```diff
@@ -28,6 +28,10 @@ func (idp *identityProviderMock) TemporaryKey(id string) (string, error) {
 		return "", manager.ErrUnauthorizedAccess
 	}
 
+	if id == "mySecretJwtToken" {
```
Comment (mijicd) [inline manager/mocks/security.go]: Unnecessary.

**Review by mijicd — CHANGES_REQUESTED** [review]

**Review by mijicd — APPROVED** [review]

**Pull request merged**